Stop/Start Model e2e Test

katieperry4 opened the pull request:
<!--- If this is a non-code change, this template is not required; reference any issues or top-level descriptions as needed -->
<!--- All code change PRs should relate to an issue, reference it here; see example below -->
<!--- https://issues.redhat.com/browse/RHOAIENG-123456 -->
Closes [RHOAIENG-27156](https://issues.redhat.com/browse/RHOAIENG-27156)

## Description
<!--- Describe your changes in detail; the what, the why, any findings, etc -->
<!--- Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->
Added an e2e test to verify the user can stop and start a model via the UI.
Steps:
- Log in
- Go to the project / make it
- Deploy a model
- Make sure it's running
- Stop the model -> the modal should pop up as well
- Wait for the model to stop
- Restart the model
- Make sure the model restarts

I also made some changes to `checkInferenceServiceState` because it was only checking if it was `loaded` or not so I added a `checkStopped` and `requireLoadedState` so if you don't need the `loaded` state (like in this case where I need to check that the model has a `stopped` status) then you don't have to check for that, just the conditions you're looking for. 
(Should not affect other tests that use `checkInferenceServiceState` I didn't want to copy and paste this util and make it work just to check stopped status)

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Tested locally and passed all jenkins testing

## Test Impact
<!--- What tests have you done to cover the implemented functionality -->
<!--- If tests are not applicable, explain why here -->
Adding cypress e2e test

To test it you can run it locally or run it on Jenkins
**Test e2e tests [locally](https://github.com/opendatahub-io/odh-dashboard/blob/73b8127e3304046c183bcc59ef80fd7e853aa507/frontend/src/__tests__/cypress/cypress/tests/e2e/README.md#running-tests):** 
1. Oc login 
2. Open one terminal 
   a. `cd frontend`
   b. `npm run start:dev:ext`
3. Open another terminal 
   a. `cd frontend`
   b. `export CY_TEST_CONFIG=/Path/to/your/test/variables/here.yml`
   c. `npm run cypress:open`
   d. Opens UI and can run tests from there


## Request review criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 
<!--- You can ignore these if you are doing manifest, backend, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **Tests**
  - Added a new end-to-end test to verify that machine learning models can be stopped and started within a data science project.
  - Introduced new test data and updated utility functions to support model stop/start state validation.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

katieperry4 commented:
@Griffin-Sullivan Totally fair! I'll keep an eye on the other one and update this once it merges 👍

Griffin-Sullivan commented:
Ok can you push your changes so we can at least see everything. To me that sounds like we aren't flipping to Started when we should be on the label which could be a bug

katieperry4 commented:
@Griffin-Sullivan checking the labels was being really flaky even when I ran it on your stopping PR. Sometimes the label switches over too quickly or sometimes it get wonky with polling weirdness. (it would report back as started and not show the `started` label.) 
So I just added:
```
kServeRow
        .findStatusLabel()
        .invoke('text')
        .should('match', /Starting|Started/);
```
and 
```
kServeRow
        .findStatusLabel()
        .invoke('text')
        .should('match', /Stopping|Stopped/);
```
for the ones being wonky (`stopping` and `started`) incase they happen too quickly or don't update fast enough

`starting` and `stopped` are consistent so I added:
`kServeRow.findStatusLabel('Stopped').should('exist');`
`kServeRow.findStatusLabel('Starting').should('exist');`

Griffin-Sullivan commented:
/test-e2e
